src/lib/mongoBackend/mongoQueryTypes.cpp

```diff
@@ -264,7 +288,20 @@ HttpStatusCode mongoEntityTypes
       }
     }
 
-    responseP->entityTypeVector.push_back(entityType);
+    // entityType corresponding to nullId case is skipped, as it is (eventually) added outside the for loop
+    if (!nullId)
+    {      
+      responseP->entityTypeVector.push_back(entityType);
+    }
+  }
+
+  if (emptyEntityTypeUsed)
+  {
+    responseP->entityTypeVector.push_back(emptyEntityType);
+  }
```

Comment:
As I thought.
You could do it inside also, with a `if (!emptyEntityTypeUsedPushed) push()`, I imagine.
However, not interesting.

NTC